features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2

```diff
@@ -456,6 +456,7 @@
         <EnableAnonymousMode>{{apim.devportal.enable_anonymous_mode}}</EnableAnonymousMode>
         <EnableCrossTenantSubscription>{{apim.devportal.enable_cross_tenant_subscriptions}}</EnableCrossTenantSubscription>
         <DefaultReservedUsername>{{apim.devportal.default_reserved_username}}</DefaultReservedUsername>
+        <CreateDefaultApplication>{{apim.devportal.create_default_application}}</CreateDefaultApplication>
```

Comment:
Let's add a config like this. This will make the configuration an optional one. if this is not define, product would work in the normal way. if we need to only desable the app, we can do it.

       {% if apim.devportal.create_default_application is defined %}
        <CreateDefaultApplication>{{apim.devportal.create_default_application}}</CreateDefaultApplication>
       {% endif %}

Reply:
Addressed via https://github.com/wso2/carbon-apimgt/pull/10498/commits/8616dc208127cc817d5ca029a43806768f6b650c